### ISSUE TITLE
lint: add "fat" value copy in range loops

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#82]
When iterating with for-range with value variable, one should be careful to avoid unexpected performance drops.

Simple example:
```go

type fat struct {
	body [1024]byte
	x    int
}

func fatCopy(xs []fat) int {
	v := 0
	for _, x := range xs {
		v += x.x
	}
	return v
}

func fatIndex(xs []fat) int {
	v := 0
	for i := range xs {
		v += xs[i].x
	}
	return v
}
```
First `fatCopy` does 1000+ bytes copy every iteration.
`fatIndex` does no copy (it reads from proper offset instead).

We should warn if values bigger than X are being iterated over.

Disasm:
```asm
"".fatCopy STEXT size=218 args=0x20 locals=0x818
	00000 (GOPATH/src/main/foo.go:8)	TEXT	"".fatCopy(SB), $2072-32
	00000 (GOPATH/src/main/foo.go:8)	MOVQ	(TLS), CX
	00009 (GOPATH/src/main/foo.go:8)	LEAQ	-1944(SP), AX
	00017 (GOPATH/src/main/foo.go:8)	CMPQ	AX, 16(CX)
	00021 (GOPATH/src/main/foo.go:8)	JLS	208
	00027 (GOPATH/src/main/foo.go:8)	SUBQ	$2072, SP
	00034 (GOPATH/src/main/foo.go:8)	MOVQ	BP, 2064(SP)
	00042 (GOPATH/src/main/foo.go:8)	LEAQ	2064(SP), BP
	00050 (GOPATH/src/main/foo.go:8)	FUNCDATA	$0, gclocals·a36216b97439c93dafebe03e7f0808b5(SB)
	00050 (GOPATH/src/main/foo.go:8)	FUNCDATA	$1, gclocals·33cdeccccebe80329f1fdbee7f5874cb(SB)
	00050 (GOPATH/src/main/foo.go:8)	MOVQ	"".xs+2088(SP), AX
	00058 (GOPATH/src/main/foo.go:8)	MOVQ	"".xs+2080(SP), CX
	00066 (GOPATH/src/main/foo.go:8)	XORL	DX, DX
	00068 (GOPATH/src/main/foo.go:8)	XORL	BX, BX
	00070 (GOPATH/src/main/foo.go:10)	JMP	179
	00072 (GOPATH/src/main/foo.go:10)	MOVQ	(CX), R8
	00075 (GOPATH/src/main/foo.go:10)	MOVQ	R8, ""..autotmp_8+1032(SP)
	00083 (GOPATH/src/main/foo.go:10)	LEAQ	8(CX), SI
	00087 (GOPATH/src/main/foo.go:10)	LEAQ	""..autotmp_8+1040(SP), DI
	00095 (GOPATH/src/main/foo.go:10)	DUFFCOPY	$0
	00114 (GOPATH/src/main/foo.go:10)	MOVQ	""..autotmp_8+1032(SP), R8
	00122 (GOPATH/src/main/foo.go:10)	MOVQ	R8, "".x(SP)
	00126 (GOPATH/src/main/foo.go:10)	LEAQ	"".x+8(SP), DI
	00131 (GOPATH/src/main/foo.go:10)	LEAQ	""..autotmp_8+1040(SP), SI
	00139 (GOPATH/src/main/foo.go:10)	DUFFCOPY	$0
	00158 (GOPATH/src/main/foo.go:10)	ADDQ	$1032, CX
	00165 (GOPATH/src/main/foo.go:10)	INCQ	DX
	00168 (GOPATH/src/main/foo.go:11)	MOVQ	"".x+1024(SP), R8
	00176 (GOPATH/src/main/foo.go:11)	ADDQ	R8, BX
	00179 (GOPATH/src/main/foo.go:10)	CMPQ	DX, AX
	00182 (GOPATH/src/main/foo.go:10)	JLT	72
	00184 (GOPATH/src/main/foo.go:13)	MOVQ	BX, "".~r1+2104(SP)
	00192 (GOPATH/src/main/foo.go:13)	MOVQ	2064(SP), BP
	00200 (GOPATH/src/main/foo.go:13)	ADDQ	$2072, SP
	00207 (GOPATH/src/main/foo.go:13)	RET
	00208 (GOPATH/src/main/foo.go:13)	NOP
	00208 (GOPATH/src/main/foo.go:8)	PCDATA	$0, $-1
	00208 (GOPATH/src/main/foo.go:8)	CALL	runtime.morestack_noctxt(SB)
	00213 (GOPATH/src/main/foo.go:8)	JMP	0

"".fatIndex STEXT nosplit size=56 args=0x20 locals=0x0
	00000 (GOPATH/src/main/foo.go:16)	TEXT	"".fatIndex(SB), NOSPLIT, $0-32
	00000 (GOPATH/src/main/foo.go:16)	FUNCDATA	$0, gclocals·a36216b97439c93dafebe03e7f0808b5(SB)
	00000 (GOPATH/src/main/foo.go:16)	FUNCDATA	$1, gclocals·33cdeccccebe80329f1fdbee7f5874cb(SB)
	00000 (GOPATH/src/main/foo.go:16)	MOVQ	"".xs+16(SP), AX
	00005 (GOPATH/src/main/foo.go:16)	MOVQ	"".xs+8(SP), CX
	00010 (GOPATH/src/main/foo.go:16)	XORL	DX, DX
	00012 (GOPATH/src/main/foo.go:16)	XORL	BX, BX
	00014 (GOPATH/src/main/foo.go:18)	JMP	45
	00016 (GOPATH/src/main/foo.go:18)	LEAQ	1(DX), SI
	00020 (GOPATH/src/main/foo.go:18)	MOVQ	DX, DI
	00023 (GOPATH/src/main/foo.go:19)	SHLQ	$10, DX
	00027 (GOPATH/src/main/foo.go:19)	LEAQ	(DX)(DI*8), DI
	00031 (GOPATH/src/main/foo.go:19)	MOVQ	1024(CX)(DI*1), DI
	00039 (GOPATH/src/main/foo.go:19)	ADDQ	DI, BX
	00042 (GOPATH/src/main/foo.go:19)	MOVQ	SI, DX
	00045 (GOPATH/src/main/foo.go:18)	CMPQ	DX, AX
	00048 (GOPATH/src/main/foo.go:18)	JLT	16
	00050 (GOPATH/src/main/foo.go:21)	MOVQ	BX, "".~r1+32(SP)
	00055 (GOPATH/src/main/foo.go:21)	RET
```

**Comment by Quasilyte** [comment]
Huh. What is even more interesting is that this check actually finds problems in Go core.
